CalibFormats/HcalObjects/src/HcalDbService.cc

```diff
@@ -223,10 +224,11 @@ bool HcalDbService::convertPedestalWidths(const HcalGenericDetId& fId,
       pedTrueWidth[i] = x;
       continue;
     }
-    // assume QIE is linear in low range and use x1=0 and x2=1
-    // y = (y2-y1) * (x) [do not add any constant, only scale!]
-    float y2 = coder->charge(*shape, 1, i);
-    float y1 = coder->charge(*shape, 0, i);
+    float y = pedestal->getValues()[i];
+    int x1 = (int)std::floor(y);
+    int x2 = (int)std::floor(y + 1.);
+    float y1 = coder->charge(*shape, x1, i);
```

Comment:
You are still using a linear approximation here rather than evaluating the charge at pedestal +/- pedestalWidth/2. Is that done to use a uniform treatment of the cases in which the pedestal width is less than one (in which case `x1` would sometimes equal `x2`) and the cases in which the pedestal width is larger than one, assuming that the linear approximation is still good enough when evaluated at the correct ADC value.

Reply:
ADC units for ped.widths is a bit tricky for high values. 
(1) we need to keep it simple and compatible with old good first-subrange evaluation, including small widths values handling for legacy HCAL, and for Phase1,2 HO and HF. 
(2) here we have kind of "unpacking" of ped.values from DB, so the most important is to keep it in sync with "packing" in the ped.analysis code. 

NB: ped.widths are used solely as a component of the pulse fit "error estimator" term, so don't need high precision, unlike pedestals (mean) themselves.

And I suppose HCAL will move at some point to linearized fC units for peds and ped.widths for Run3. It's already the case for (yet hardcoded) conditions for Phase2.